tests/test_app/TestApp/Controller/TestsAppsController.php

```diff
@@ -62,9 +60,7 @@ public function redirect_to_permanent()
 
     public function set_type()
     {
-        $this->response->type('json');
-
-        return $this->response;
+        return $this->response->withType('json');
```

Comment:
is this true?
```php
       $this->response->type('json');

        return $this->response;
```
will change origin `response` and returns (mutable )
```php
$this->response->withType('json');
```
will no change origin `response` and returns (immutable )


I mean

```php
// value of $this->response
$responseResult=set_type()

$this->response == $responseResult //before
$this->response != $responseResult //now
```

I know this is just a testCase

Reply:
You're right this doesn't modify the controller's `response` property. In this situation that isn't relevant as a controller action returning a response overrides the property.